lldb/tools/debugserver/source/MacOSX/MachProcess.mm

```diff
@@ -1594,6 +1594,10 @@ static uint64_t bits(uint64_t value, uint32_t msbit, uint32_t lsbit) {
                          m_sent_interrupt_signo);
       }
     } else {
+      // We've requested that the process stop anew; if we had recorded this
+      // requested stop as being in place when we resumed (& therefore would
+      // throw it away), clear that.
+      m_auto_resume_signo = 0;
```

Comment:
Ok, and this is protected by the `m_exception_and_signal_mutex`. 👍

Reply:
Yeah, you added that lock when you were addressing TSAN issues, it's definitely required.